### DISCUSSION TITLE
Lesson 6: cast send ... not storing the input here: "store(uint256)" 123 ...

### DISCUSSION THREAD

**Discussion by Mohsen2083** [Cyfrin/foundry-full-course-cu#1203] [Q&A]
Hello, I am having trouble figuring out why doesn't my code store the value after typing "cast send $PUBLIC_KEY "store(uint256)" 123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY" in my terminal, meaning after I do the cast send, if I type "cast call $PUBLIC_KEY "retrieve()" --rpc-url $RPC_URL --private-key $PRIVATE_KEY", I do not get the 0x1234... thing back, that would be awesome if someone could help me :)
![Screenshot 2023-12-14 103506](https://github.com/Cyfrin/foundry-full-course-f23/assets/131712492/0df1ebc1-04bf-4b04-b952-31778fd46251)

**Comment by sivajisj** [comment]
1.Make sure that the .env file is in the correct location and contains valid environment variable assignments. After sourcing the file, the environment variables should be available for the current shell session.
```bash

source .env

```


2.Can you provide the error you are encountering?

**Reply by Mohsen2083** [reply]
I am not getting an error but when i try to type "cast call $PUBLIC_KEY "retrieve()" --rpc-url $RPC_URL --private-key $PRIVATE_KEY" inside my terminal the 123 value which I stored via store function does not show up

**Comment by sivajisj** [comment]
3.Please share your repository; I will check to see if it works fine.

**Reply by Mohsen2083** [reply]
Hi, thx for your reply, here it is:
https://github.com/Mohsen2083/Lesson-6/tree/3f41f4a0a4fe4759c2f17eb17c5bedfbb3064d24/foundry-f23

**Reply by Verifieddanny** [reply]
Having the same issues as well

**Comment by Amrit-ko** [comment]
Hello @Mohsen2083. It seems that you're sending a transaction to a contract that doesn't exist 
Try to do this steps:
1. launch anvil
2. in new terminal run ```forge script script/DeploySimpleStorage.s.sol --rpc-url $RPC_URL --broadcast --private-key $PRIVATE_KEY```
3. check if contract address matches your $PUBLIC_KEY, if not change it to the correct one
4. run ```cast send $PUBLIC_KEY "store(uint256)" 123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY```
5. run ```cast call $PUBLIC_KEY "retrieve()" --rpc-url $RPC_URL --private-key $PRIVATE_KEY```
as you using .env don`t forget to run ```source .env``` before running commands

**Reply by Mohsen2083** [reply]
Hello, thanks for your detailed reply, I just did everything you said and still having the same issue😿😿😿

**Reply by Mohsen2083** [reply]
I am sure my .env file is working fine btw.

**Reply by Verifieddanny** [reply]
try this

```
cast send 0x5FbDB2315678afecb367f032d93F642f64180aa3 "store(uint256)"  123 --rpc-url $RPC_URL --private-key $PRIVATE_KEY
```

then to read(call): if you're using anvil:
```
cast call 0x5FbDB2315678afecb367f032d93F642f64180aa3 "recieve()"
```

then to read(call): if you're not using anvil:
```
cast call 0x5FbDB2315678afecb367f032d93F642f64180aa3 "recieve()" --rpc-url $RPC_URL
```

you can decide to use the CA in the given syntax

**Reply by Mohsen2083** [reply]
still only getting "0x" and nothing else haha

**Reply by knaftoli** [reply]
This helped me:
[https://github.com/Cyfrin/foundry-full-course-f23/discussions/459#discussioncomment-6553968](url)
